src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2174,6 +2180,30 @@ private void HandleNewRequest(int node, BuildRequestBlocker blocker)
             PerformSchedulingActions(response);
         }
 
+        /// <summary>
+        /// Handles a resource request coming from a node.
+        /// </summary>
+        private void HandleResourceRequest(int node, ResourceRequest request)
+        {
+            if (request.IsResourceAcquire)
+            {
+                // Resource request requires a response and may be blocking. Our continuation is effectively a callback
+                // to be called once at least one core becomes available.
```

Comment:
The tasks that opt in must cooperate and release resources they don't use any longer. It would be good though to have logging (if not already)  of all RequestCores/Release calls with the tasks name and the number of cores requested/aquired or released.

Reply:
I have added logging as suggested.